src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -217,6 +324,18 @@ public String getDartSdkPath() throws ExecutionException {
     return FlutterSdkUtil.pathToDartSdk(getHomePath());
   }
 
+  /**
+   * Returns true if the Dart SDK has been downloaded for this Flutter SDK.
+   */
+  public boolean hasDartSdk() {
+    VirtualFile file = LocalFileSystem.getInstance().findFileByPath(myHomePath + "/bin/cache/dart-sdk");
```

Comment:
Aside: we use `"/bin/cache/dart-sdk"` in a few places --- maybe a good idea to stuff into a constant?